### PR TITLE
added kwargs to fakedataset

### PR DESCRIPTION
**Pull request by arunppsg**
Fix #4067. Changes made in this PR allows users to add additional features which can be generated via `FakeDataset`.

### CODE REVIEW

File: torch_geometric/datasets/fake.py [hunk 1]
```diff
@@ -36,20 +36,14 @@ class FakeDataset(InMemoryDataset):
...
-        is_undirected: bool = True,
-        transform: Optional[Callable] = None,
-        pre_transform: Optional[Callable] = None,
-    ):
+    def __init__(self, num_graphs: int = 1, avg_num_nodes: int = 1000,
+                 avg_degree: int = 10, num_channels: int = 64,
+                 edge_dim: int = 0, num_classes: int = 10, task: str = "auto",
+                 is_undirected: bool = True,
+                 transform: Optional[Callable] = None,
+                 pre_transform: Optional[Callable] = None, **kwargs):
```
Comment (rusty1s) [inline torch_geometric/datasets/fake.py:46]:
```suggestion
                 pre_transform: Optional[Callable] = None, **kwargs,):
```

File: torch_geometric/datasets/fake.py [hunk 2]
```diff
@@ -36,20 +36,14 @@ class FakeDataset(InMemoryDataset):
             an :obj:`torch_geometric.data.Data` object and returns a
             transformed version. The data object will be transformed before
             every access. (default: :obj:`None`)
+        **kwargs (optional): Additional attributes and their shapes.
```
Comment (rusty1s) [inline torch_geometric/datasets/fake.py:39]: Can we give an example here? For example,
```
**kwargs (optional): Additional attributes and their shapes, *e.g.*, :obj:`global_feat=(1, 128)`
```

File: torch_geometric/datasets/fake.py [hunk 3]
```diff
@@ -64,10 +58,12 @@ def __init__(
...
         self.is_undirected = is_undirected
 
-        data_list = [self.generate_data() for _ in range(max(num_graphs, 1))]
+        data_list = [
+            self.generate_data(**kwargs) for _ in range(max(num_graphs, 1))
+        ]
         self.data, self.slices = self.collate(data_list)
 
-    def generate_data(self) -> Data:
+    def generate_data(self, **kwargs) -> Data:
```
Comment (rusty1s) [inline torch_geometric/datasets/fake.py:66]:
```suggestion
    def generate_data(self) -> Data:
```

File: torch_geometric/datasets/fake.py [hunk 4]
```diff
@@ -93,6 +89,9 @@ def generate_data(self) -> Data:
         elif self.edge_dim == 1:
             data.edge_weight = torch.rand(data.num_edges)
 
+        for feature_name, feature_shape in kwargs.items():
```
Comment (rusty1s) [inline torch_geometric/datasets/fake.py:92]:
```suggestion
        for feature_name, feature_shape in self.kwargs.items():
```
Let's use `kwargs` as a property (similar to all other variables).

**Review by rusty1s — APPROVED** [review]
LGTM. Thank you! Left a few comments.

File: torch_geometric/datasets/fake.py [hunk 5]
```diff
@@ -36,6 +36,7 @@ class FakeDataset(InMemoryDataset):
             an :obj:`torch_geometric.data.Data` object and returns a
             transformed version. The data object will be transformed before
             every access. (default: :obj:`None`)
+        **kwargs (optional): Additional attributes and their shapes *e.g.* :obj:`{'global_features':5}`.
```
Comment (rusty1s) [inline torch_geometric/datasets/fake.py]: This is still a keyword arg, right? So `global_feature=5` should be correct.

Comment (arunppsg) [reply torch_geometric/datasets/fake.py]: Yeah, I agree.

**Comment by arunppsg** [comment]
> Thanks for the updates. One minor thing: Can we add a test for this as well?

Yeah, I was also thinking about the same. Can we overload `test_fake_dataset` or create a new one `test_fake_dataset_kwargs`? We can check for the shape of attributes.

**Comment by rusty1s** [comment]
I think you can just add the test in the already existing test method :)

**Comment by arunppsg** [comment]
Thanks for your suggestions, I added tests to the fakedataset.

**Comment by rusty1s** [comment]
Awesome, thank you.

**Pull request merged**